fix: fixes server-side diff with Helm+Kubernetes (#28092)
argoproj/argo-cd: docs/operator-manual/upgrading/3.5-3.6.md

- If you set `ARGOCD_REPO_OTLP_HEADERS` directly (for example via a custom Deployment patch or Helm
  values), rename it to `ARGOCD_REPO_SERVER_OTLP_HEADERS`. The old variable is no longer read.

### Default local include paths now include specific Helm files under `charts/`

When using `argocd app diff --local --server-side-generate`, the implicit default for `--local-include` has changed from `["*.yaml", "*.yml", "*.json"]` to `["*.yaml", "*.yml", "*.json", "charts/**/*.yaml", "charts/**/*.yml", "charts/**/*.tpl"]`.

This means that common Helm chart files (YAML and TPL) under a local `charts/` subdirectory will now be sent to the repo-server for server-side manifest generation. Other files under `charts/` (such as large binary assets or sensitive non-YAML files) are excluded by default to avoid unintentional transmission.

> [!WARNING]

> > If your application source contains non-YAML content under `charts/` that you do not want transmitted, you must override the include set via repeated `--local-include` flags to narrowly specify what should be sent.

Reviewer: ```suggestion
> If your application source contains non-YAML content under `charts/` that you want transmitted, you must override the include set via repeated `--local-include` flags to extend what should be sent, including
   `["*.yaml", "*.yml", "*.json", "charts/**/*.yaml", "charts/**/*.yml", "charts/**/*.tpl"]` 
   if you need them.
```
